includes/core-changes/aspnetcore/3.0/caching-response-pubternal-to-internal.md

```diff
@@ -0,0 +1,66 @@
+### Caching: ResponseCaching "pubternal" types changed to internal
+
+In ASP.NET Core 3.0, "pubternal" types in `ResponseCaching` have been changed to `internal`.
+
+In addition, default implementations of `IResponseCachingPolicyProvider` and `IResponseCachingKeyProvider` are no longer added to services as part of the `AddResponseCaching` method.
+
+#### Change description
+
+In ASP.NET Core, "pubternal" types are declared as `public` but reside in a namespace suffixed with `.Internal`. While these types are public, they have no support policy and are subject to breaking changes. Unfortunately, accidental use of these types has been common, resulting in breaking changes to these projects and limiting the ability to maintain the framework.
+
+#### Version introduced
+
+3.0
+
+#### Old behavior
+
+These types were publicly visible, but unsupported.
+
+#### New behavior
+
+These types are now `internal`.
+
+#### Reason for change
+
+The `internal` scope better reflects the unsupported policy.
+
+#### Recommended action
+
+Copy types that are used by your app or library.
+
+#### Category
+
+ASP.NET Core
+
+#### Affected APIs
+
+- <xref:Microsoft.AspNetCore.ResponseCaching.Internal.CachedResponse?displayProperty=fullName>
```

Comment:
The internal APIs are being removed from publishing so these links will no longer work.

You might just do the following for the ones showing up in the build report.

```suggestion
- `Microsoft.AspNetCore.ResponseCaching.Internal.CachedResponse`
```